core/src/main/java/edu/wpi/grip/core/operations/composite/PublishVideoOperation.java

```diff
@@ -160,7 +171,16 @@ public void perform() {
       throw new IllegalArgumentException("Input image must not be empty");
     }
 
-    copyJavaCvToOpenCvMat(input, publishMat);
+    // "copy" the input data to an OpenCV mat for cscore to use
+    // This basically just wraps the mat pointer in a different wrapper object
+    // No copies are performed, but it means we have to be careful about making sure we use the
+    // same version of JavaCV and OpenCV to minimize the risk of binary incompatibility.
+    // This copy only needs to happen once, since the operation input image is always the same
+    // object that gets copied into.  The data address will change, however, if the image is resized
+    // or changes type.
+    if (publishMat == null || publishMat.nativeObj != input.address()) {
+      publishMat = new Mat(input.address());
```

Comment:
Can you keep the test you had for `copyJavaCvToOpenCvMat ` arround for making sure that we know if the memory models are incompatible when we bump versions in the future?

Reply:
Sure.